sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -119,6 +120,18 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
         limitSQL = limit.sql
       } yield s"$childSQL LIMIT $limitSQL"
 
+    // TABLESAMPLE is part of tableSource clause in the parser,
+    // and thus we must handle it with subquery.
+    case Sample(lb, ub, withReplacement, _, child @ Subquery(alias, grandChild))
```

Comment:
Agree with @rxin. Especially, I don't think `MetastoreRelation` is wrapped by a `Subquery` in this case.